runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/RunProcessor.java

```diff
@@ -34,18 +27,18 @@ public RunProcessor(PlatformType platformType, String name) {
     this.name = name;
   }
 
-  public String getName() {
+  public final String getName() {
     return name;
   }
 
-  public PlatformType getPlatformType() {
+  public final PlatformType getPlatformType() {
     return platformType;
   }
 
   public Stream<File> getRunsFromRoot(File root) {
     return Arrays.stream(root.listFiles(File::isDirectory));
   }
 
-  public abstract NotificationDto process(File runDirectory, TimeZone tz) throws IOException;
+  public abstract NotificationDto process(File runDirectory, TimeZone tz) throws Exception;
```

Comment:
I do not understand this rationale, but fine. The basic policy we have built is convert all checked exceptions to runtime exceptions rather than just allow checked exceptions to propagate.